app/src/features/apiClient/helpers/APIClientManager/index.ts

```diff
@@ -15,6 +22,22 @@ export const executeAPIRequest = async (
   requestCollectionId?: string
 ): Promise<RQAPI.Entry | RQAPI.RequestErrorEntry> => {
   // Process request configuration with environment variables
+
+  if (!isEmpty(entry.auth)) {
+    const authType = entry.auth.currentAuthType;
+    const requestHeaders = entry.request.headers;
+    const queryParmas = entry.request.queryParams;
+    const addToQuery = authType === AUTHORIZATION_TYPES.API_KEY && entry.auth[authType].addTo === "QUERY";
+
+    if (addToQuery) {
+      entry.request.headers = deleteAuthOptions(requestHeaders);
```

Comment:
Why do we need to deleteAuthOptions first? We never added them during current execution cycle.

Reply:
Case -> Added authorization api key to headers clicked on send then rendered request adds it to headers if user then changes from headers-> query params then we have to delete auth options from headers.